src/codewhisperer/vue/backend.ts

```diff
@@ -0,0 +1,147 @@
+                editor.selection = new vscode.Selection(endOfDocument, endOfDocument)
+            })
+        })
+    }
+
+    //This function returns the OS type of the machine used in Shortcuts and Generate Suggestion Sections
+    public getOSType(): OSType {
+        return os.platform() === 'darwin' ? 'Mac' : 'RestOfOS'
+    }
+
+    //This function opens the Keyboard shortcuts in VSCode
+    async openShortCuts(): Promise<void> {
+        vscode.commands.executeCommand('workbench.action.openGlobalKeybindings', 'codewhisperer')
+    }
+
+    //This function opens the Feedback CodeWhisperer page in the webview
+    async openFeedBack(): Promise<void> {
+        vscode.commands.executeCommand('aws.submitFeedback', 'CodeWhisperer')
+    }
+
+    //------Telemetry------
+    /** This represents the cause for the webview to open, whether a certain button was clicked or it opened automatically */
+    #codeWhispererSource?: CodeWhispererSource
+
+    setSource(source: CodeWhispererSource | undefined) {
+        if (this.#codeWhispererSource) {
+            return
+        }
+        this.#codeWhispererSource = source
+    }
+
+    emitUiClick(id: CodeWhispererUiClick) {
+        telemetry.ui_click.emit({
+            elementId: id,
+        })
+    }
+    //Telemetry for CodeWhisperer Try Example with two params Language and Task Type
+    emitTryExampleClick(languageSelected: CodewhispererLanguage, taskType: CodewhispererGettingStartedTask) {
+        telemetry.codewhisperer_onboardingClick.emit({
+            codewhispererLanguage: languageSelected,
+            codewhispererGettingStartedTask: taskType,
+        })
+    }
+}
+// export type codewhispererLanguage = 'python' | 'javascript' | 'csharp'
+//List of all events that are emitted from the webview of CodeWhisperer
+export type CodeWhispererUiClick =
+    | 'codewhisperer_Resources_Documentation'
+    | 'codewhisperer_Resources_Feedback'
+    | 'codewhisperer_Workshop_LearnMore'
+    | 'codewhisperer_Commands_KeyboardShortcutsEditor'
+    | 'codewhisperer_ScanCode_LearnMore'
+    | 'codewhisperer_GenerateSuggestions_LearnMore'
+    | 'codewhisperer_Learn_PageOpen'
+
+const Panel = VueWebview.compilePanel(CodeWhispererWebview)
+let activePanel: InstanceType<typeof Panel> | undefined
+let subscriptions: vscode.Disposable[] | undefined
+
+export type CodeWhispererSource = 'codewhispererDeveloperTools'
+
+// This function is called when the extension is activated : Webview of CodeWhisperer
+export async function showCodeWhispererWebview(
+    ctx: vscode.ExtensionContext,
+    source: CodeWhispererSource,
+    start: string
+): Promise<void> {
+    activePanel ??= new Panel(ctx, start) // "start" Parameter is passed to the constructor of CodeWhispererWebview to seperate the user experience from first time user to regualr signIn user
+    activePanel.server.setSource(source)
+
+    const webview = await activePanel!.show({
```

Comment:
Does `activePanel!` assume/assert it needs to be non-null/not undefined? I would do an early return in case it's undefined instead of doing `activePanel!` here.